pymc3/distributions/continuous.py

```diff
@@ -3268,12 +3268,23 @@ def logp(self, value):
         sigma = self.sigma
         nu = self.nu
 
-        # This condition suggested by exGAUS.R from gamlss
-        lp = tt.switch(tt.gt(nu,  0.05 * sigma),
-                       - tt.log(nu) + (mu - value) / nu + 0.5 * (sigma / nu)**2
-                       + logpow(std_cdf((value - mu) / sigma - sigma / nu), 1.),
-                       - tt.log(sigma * tt.sqrt(2 * np.pi))
-                       - 0.5 * ((value - mu) / sigma)**2)
+        # This condition is suggested by exGAUS.R from gamlss
+        lp = tt.switch(
+            tt.gt(nu, 0.05 * sigma),
+            -tt.log(nu)
+            + (mu - value) / nu
+            + 0.5 * (sigma / nu) ** 2
+            + logpow(
+                tt.switch(
+                    tt.eq(std_cdf((value - mu) / sigma - sigma / nu), 0),
+                    np.finfo(float).eps,
+                    std_cdf((value - mu) / sigma - sigma / nu)
+                ),
+                1.0
+            ),
+            -tt.log(sigma * tt.sqrt(2 * np.pi)) - 0.5 * ((value - mu) / sigma) ** 2,
+        )
```

Comment:
```suggestion
        standardized_val = (value - mu) / sigma
        cdf_val = std_cdf(standardized_val - sigma / nu)
        cdf_val_safe = tt.switch(
            tt.eq(cdf_val, 0), np.finfo(pm.floatX).eps, cdf_val)
        lp = tt.switch(
            tt.gt(nu, 0.05 * sigma),
            -tt.log(nu) + (mu - value) / nu + 0.5 * (sigma / nu) ** 2 + logpow(cdf_val_safe, 1.0),
            -tt.log(sigma * tt.sqrt(2 * np.pi)) - 0.5 * standardized_val ** 2,
        )
```

Reply:
Just one note: when I use `np.finfo(pm.floatX).eps` instead of `np.finfo(float).eps`, it raises `ValueError: data type <class 'numpy.object_'> not inexact`. Is it important to use our float type?
Full traceback:
```python
Traceback (most recent call last):
  File "/Users/alex_andorra/opt/anaconda3/envs/pymc-dev/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3319, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-3-cf016555abd8>", line 1, in <module>
    pm.ExGaussian.dist(0., .25, 1. / 6).logp(y).eval().round(1)
  File "/Users/alex_andorra/tptm_alex/pymc3/pymc3/distributions/continuous.py", line 3344, in logp
    cdf_val_safe = tt.switch(tt.eq(cdf_val, 0), np.finfo(floatX).eps, cdf_val)
  File "/Users/alex_andorra/opt/anaconda3/envs/pymc-dev/lib/python3.8/site-packages/numpy/core/getlimits.py", line 381, in __new__
    raise ValueError("data type %r not inexact" % (dtype))
ValueError: data type <class 'numpy.object_'> not inexact
```

Reply:
I think ideally we use the float type that consistent with the rest of the pm.Model - IIUC user set the global dtype and that should reflect in the `pm.floatX`?

Reply:
I think it just converts to theano floatX type:
```python
def floatX(X):
    """
    Convert a theano tensor or numpy array to theano.config.floatX type.
    """
    try:
        return X.astype(theano.config.floatX)
    except AttributeError:
        # Scalar passed
        return np.asarray(X, dtype=theano.config.floatX)
```

Reply:
Ok, I think this does what you're talking about: `np.finfo(theano.config.floatX)`
Pushing, and if it's ok for you we can merge 👌